pkg/provisioner/ironic/ironic.go

```diff
@@ -1205,6 +1216,28 @@ func (p *ironicProvisioner) PowerOn() (result provisioner.Result, err error) {
 func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 	p.log.Info("ensuring host is powered off")
 
+	result, err = p.softPowerOff()
+	if err != nil {
+		switch err.(type) {
+		// In case of soft power off is unsupported or has failed,
+		// we activate hard power off.
+		case SoftPowerOffUnsupportedError, SoftPowerOffFailed:
+			return p.hardPowerOff()
+		case HostLockedError:
+			return p.softPowerOff()
```

Comment:
I don't think we want to call again straight away in this case, but rather requeue with a delay so we can try again.

Reply:
```
		result, err = p.changePower(ironicNode, nodes.SoftPowerOff)
		if err != nil {
			result.RequeueAfter = powerRequeueDelay
			return result, err
		}
```
In the `softPowerOff `function, when these errors occur, `powerRequeueDelay` will be added, is it necessary to add additional delay?

Reply:
That's a delay after calling it a second time. So you call it twice in succession, then wait to go around again. It's extremely unlikely that the second call will succeed if the first didn't, so it's a waste. Just requeue with a delay.

Reply:
Updated, before calling `p.softPowerOff ()`, I added a `sleep` function. PTAL

Reply:
Sleeping blocks the controller from reconciling any other resources. Just return, we'll get back here soon enough.

Reply:
thanks, updated, PTAL.

Reply:
@zaneb PTAL